From 17acda376434e84cf462e707651fab7bfbf02549 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <grinberg@compulab.co.il>
Date: Sun, 24 Feb 2013 12:54:05 +0200
Subject: ARM: i.MX6: cm-fx6: fix I2C3 pinmux

cm-fx6 uses GPIO_3 for I2C3_SCL and GPIO_6 for I2C3_SDA.
Configure the right pins.

Signed-off-by: Igor Grinberg <grinberg@compulab.co.il>
---
 arch/arm/mach-mx6/board-cm-fx6-dl.h |   14 ++------------
 arch/arm/mach-mx6/board-cm-fx6-q.h  |   13 ++-----------
 2 files changed, 4 insertions(+), 23 deletions(-)

diff --git a/arch/arm/mach-mx6/board-cm-fx6-dl.h b/arch/arm/mach-mx6/board-cm-fx6-dl.h
index 287f5eb..d28e955 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-dl.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-dl.h
@@ -42,10 +42,6 @@ static iomux_v3_cfg_t mx6dl_arm2_pads[] = {
 	MX6DL_PAD_RGMII_RD3__ENET_RGMII_RD3,
 	MX6DL_PAD_RGMII_RX_CTL__ENET_RGMII_RX_CTL,
 
-	/* MCLK for csi0 */
-	MX6DL_PAD_GPIO_0__CCM_CLKO,
-	MX6DL_PAD_GPIO_3__CCM_CLKO2,
-
 	/* SD1 */
 	MX6DL_PAD_SD1_CLK__USDHC1_CLK,
 	MX6DL_PAD_SD1_CMD__USDHC1_CMD,
@@ -146,17 +142,11 @@ static iomux_v3_cfg_t mx6dl_arm2_pads[] = {
 
 	/* USBOTG ID pin */
 	MX6DL_PAD_GPIO_1__USBOTG_ID,
-
-	/* MLB150 */
-	MX6DL_PAD_GPIO_3__MLB_MLBCLK,
-	MX6DL_PAD_GPIO_6__MLB_MLBSIG,
-	MX6DL_PAD_GPIO_2__MLB_MLBDAT,
-
 };
 
 static iomux_v3_cfg_t mx6dl_arm2_i2c3_pads[] = {
-	MX6DL_PAD_GPIO_5__I2C3_SCL,
-	MX6DL_PAD_GPIO_16__I2C3_SDA,
+	MX6DL_PAD_GPIO_3__I2C3_SCL,
+	MX6DL_PAD_GPIO_6__I2C3_SDA,
 };
 
 static iomux_v3_cfg_t mx6dl_arm2_spdif_pads[] = {
diff --git a/arch/arm/mach-mx6/board-cm-fx6-q.h b/arch/arm/mach-mx6/board-cm-fx6-q.h
index f436918..190eaed 100644
--- a/arch/arm/mach-mx6/board-cm-fx6-q.h
+++ b/arch/arm/mach-mx6/board-cm-fx6-q.h
@@ -42,10 +42,6 @@ static iomux_v3_cfg_t mx6q_arm2_pads[] = {
 	MX6Q_PAD_RGMII_RD3__ENET_RGMII_RD3,
 	MX6Q_PAD_RGMII_RX_CTL__ENET_RGMII_RX_CTL,
 
-	/* MCLK for csi0 */
-	MX6Q_PAD_GPIO_0__CCM_CLKO,
-	MX6Q_PAD_GPIO_3__CCM_CLKO2,
-
 	/* SD1 */
 	MX6Q_PAD_SD1_CLK__USDHC1_CLK,
 	MX6Q_PAD_SD1_CMD__USDHC1_CMD,
@@ -146,16 +142,11 @@ static iomux_v3_cfg_t mx6q_arm2_pads[] = {
 
 	/* USBOTG ID pin */
 	MX6Q_PAD_GPIO_1__USBOTG_ID,
-
-	/* MLB150 */
-	MX6Q_PAD_GPIO_3__MLB_MLBCLK,
-	MX6Q_PAD_GPIO_6__MLB_MLBSIG,
-	MX6Q_PAD_GPIO_2__MLB_MLBDAT,
 };
 
 static iomux_v3_cfg_t mx6q_arm2_i2c3_pads[] = {
-	MX6Q_PAD_GPIO_5__I2C3_SCL,
-	MX6Q_PAD_GPIO_16__I2C3_SDA,
+	MX6Q_PAD_GPIO_3__I2C3_SCL,
+	MX6Q_PAD_GPIO_6__I2C3_SDA,
 };
 
 static iomux_v3_cfg_t mx6q_arm2_spdif_pads[] = {
-- 
1.7.9.5

